### ISSUE TITLE
cluster_split(method="dbscan") silently ignores train_size: default eps=0.5 makes nearly everything noise

### ISSUE THREAD

**Issue by gxlarson** [gxlarson/splytters#67]
`cluster_split(method="dbscan")` returns a degenerate split, without any warning, on any data whose pairwise distances are large next to DBSCAN's default `eps=0.5`. That covers most unnormalized embeddings. A few near-duplicates are enough to produce a train set of about 10 samples, regardless of `train_size`.

Found while benchmarking the splitters on 512-d CLIP ViT-B/32 embeddings of RVL-CDIP (median row L2 norm 10.4) at `main` @ `ed2f0e7`:

| n | DBSCAN result | `cluster_split(X, 0.8, method="dbscan")` |
|---|---|---|
| 8,000 | 0 clusters, 100% noise | 6,400 / 1,600, but a **random** split (seeded fallback, `UserWarning` only) |
| 16,000 | 2 clusters (18 points), 99.9% noise | **18 train / 15,982 test**, no warning |
| 32k–256k | same pattern | test = 99.9% of n |

## Cause

- `cluster_at` (`adversarial.py:618`) builds `DBSCAN(**cluster_kwargs)`, so the default absolute `eps=0.5` applies regardless of the data's scale.
- `_assign_by_size` (`adversarial.py:46–62`) sends the noise label `-1` to test as one block.
- When nearly everything is noise:
  - If there are **no** clusters, train is empty. The fallback at `adversarial.py:700` fires and returns an exact-size random split. The result is not adversarial, and the only signal is a `UserWarning`.
  - If there are **any** clusters (a handful of duplicates is enough), both sides are non-empty. The fallback does not fire, and test ends up being almost all of the data.
- In `centroid`, `closest` and `subset_sum`, `-1` is an ordinary cluster, so the noise block lands whole on one side or the other. With `closest` the repro below gives the mirror image, 3,999 train / 11 test.

## Repro

```python
import numpy as np, splytters as s
from sklearn.datasets import make_blobs

X, _ = make_blobs(n_samples=4000, n_features=64, centers=16, random_state=0)

s.cluster_split(X, 0.8, method="dbscan")           # 3200/800, random fallback + UserWarning

Xd = np.vstack([X, np.repeat(X[:1], 10, axis=0)])  # add 10 duplicate rows
for strategy in ("size", "centroid", "closest"):
    tr, te = s.cluster_split(Xd, 0.8, method="dbscan", strategy=strategy)
    print(strategy, len(tr), len(te))
# size 11 3999
# centroid 11 3999
# closest 3999 11     (no warnings in any of the three)
```

## Why it matters

`train_size` is silently ignored, and the split looks valid: disjoint, covering all samples, no exception. Real document and image datasets almost always contain some near-duplicates, so the "18 / 15,982" case is the typical outcome, not a corner case.

## Suggestions

1. Warn or raise when the realized train fraction differs from `train_size` by more than a tolerance. This would also catch the same whole-cluster failure with KMeans.
2. Warn when the DBSCAN noise fraction is high (for example > 50%). The fallback warning does not say that the cause is `eps`.
3. Make `eps` required for `method="dbscan"`, or derive a scale-aware default, for example a quantile of the k-th nearest-neighbour distance with k = `min_samples`.
4. Treat noise consistently across strategies. Assigning noise points individually to fill the target would respect `train_size`.
